packages/core/src/html/NodeProcessor.ts

```diff
@@ -353,7 +405,9 @@ class NodeProcessor {
         });
         mainHtmlNodes.forEach(d => NodeProcessor._trimNodes(d));
```

Comment:
Looks like there may be some typing issue here? `mainHtmlNodes` here has the type `(DomElement | DomElement[])[]` which makes `d` have the type `DomElement | DomElement[]`. `_trimNodes` however only accepts a `DomElement` as param.

Also seems like `_trimNodes` method will have some issues if an array of `DomElement`s is passed to it.